src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-snapshots-list/cephfs-subvolume-snapshots-form/cephfs-subvolume-snapshots-form.component.html

```diff
@@ -0,0 +1,99 @@
+<cd-modal [modalRef]="activeModal">
+  <ng-container i18n="form title"
+                class="modal-title">{{ action | titlecase }} {{ resource | upperFirst }}</ng-container>
+
+  <ng-container class="modal-content"
+                *cdFormLoading="loading">
+    <form name="snapshotForm"
+          #formDir="ngForm"
+          [formGroup]="snapshotForm"
+          novalidate>
+      <div class="modal-body">
+        <div class="form-group row">
+          <label class="cd-col-form-label required"
+                 for="snapshotName"
+                 i18n>Name</label>
+          <div class="cd-col-form-input">
+            <input class="form-control"
+                   type="text"
+                   placeholder="Snapshot name..."
+                   id="snapshotName"
+                   name="snapshotName"
+                   formControlName="snapshotName"
+                   autofocus>
+            <span class="invalid-feedback"
+                  *ngIf="snapshotForm.showError('snapshotName', formDir, 'required')"
+                  i18n>This field is required.</span>
+            <span class="invalid-feedback"
+                  *ngIf="snapshotForm.showError('snapshotName', formDir, 'notUnique')"
+                  i18n>The subvolume already exists.</span>
+            <span *ngIf="snapshotForm.showError('snapshotName', formDir, 'pattern')"
+                  class="invalid-feedback"
+                  i18n>Subvolume name can only contain letters, numbers, '.', '-' or '_'</span>
```

Comment:
```suggestion
            <span class="invalid-feedback"
                  *ngIf="snapshotForm.showError('snapshotName', formDir, 'notUnique')"
                  i18n>The Snapshot name already exists.</span>
            <span *ngIf="snapshotForm.showError('snapshotName', formDir, 'pattern')"
                  class="invalid-feedback"
                  i18n>Snapshot name can only contain letters, numbers, '.', '-' or '_'</span>
```

Also I don't see the validation for this errors on the `.ts` file, so not sure if it's planned to include those since it is allowed to rewrite the snapshot

Reply:
added the async validator for unique checking. the pattern one i think is not needed because i was able to create with random names i provided for testing, hence removed it